boards/st/nucleo_n657x0_q/board.cmake

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2024 STMicroelectronics
+
+if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Windows)
+    set(SIGNING_TOOL STM32_SigningTool_CLI.exe)
+else()
+    set(SIGNING_TOOL STM32_SigningTool_CLI)
+endif()
+
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+  COMMAND ${SIGNING_TOOL}
+  -bin ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
+  -nk -of 0x80000000 -t fsbl -hv 2.3
+  -o ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}_signed.bin
+  -dump ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}_signed.bin
+  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+)
+
+set_property(TARGET runners_yaml_props_target PROPERTY bin_file zephyr_signed.bin)
```

Comment:
what is this and why is this here? This should not be in a board

Reply:
STM32N6 requires binaries to be signed before flashing them. Otherwise they won't be executed.
This code is used to call the STM32_SigningTool_CLI utility to sign the binary as the target expects it.
This operation is necessary but if its place is elsewhere we can move it.

Reply:
this should be a soc thing, and it doesn't sign the hex file so that's wrong plus will completely fail if bin output is disabled in Kconfig. Would also be nice to match mcuboot signing name i.e. `<kernel>.signed.bin`

Reply:
Moving it to soc and renaming the signed binary works, but unfortunately we have no choice but to sign the bin file because the signing tool requires it:
`Error: binary file is not .bin or .stm32 extension`

Reply:
Moved the code to soc and renamed to <kernel>.signed.bin. But still using bin file because we have to.